crates/bin/pcli/src/command/query/validator.rs

```diff
@@ -155,60 +164,73 @@ impl ValidatorCmd {
-                    .iter()
-                    .map(|info| &info.validator)
-                    .find(|v| v.identity_key == identity_key)
-                    .cloned()
-                    .ok_or_else(|| anyhow::anyhow!("Could not find validator {}", identity_key))?
-                    .into();
-
+                    .into_inner();
+
+                // Coerce the validator information into TOML, or return an error if it was not
+                // found within the client's response.
+                let serialize = |v| toml::to_string_pretty(&v).map_err(Error::from);
+                let toml = validator_info
+                    .ok_or_else(|| anyhow!("response did not include validator info"))?
+                    .try_into()
+                    .context("parsing validator info")
+                    .map(|Info { validator, .. }| validator)
+                    .map(ValidatorToml::from)
+                    .and_then(serialize)?;
+
+                // Write to a file if an output file was specified, otherwise print to stdout.
                 if let Some(file) = file {
                     File::create(file)
                         .with_context(|| format!("cannot create file {file:?}"))?
-                        .write_all(toml::to_string_pretty(&validator)?.as_bytes())
+                        .write_all(toml.as_bytes())
                         .context("could not write file")?;
                 } else {
-                    println!("{}", toml::to_string_pretty(&validator)?);
+                    println!("{}", toml);
                 }
             }
+            ValidatorCmd::Status { identity_key } => {
+                // Parse the identity key and construct the RPC request.
+                let request = tonic::Request::new(GetValidatorInfoRequest {
+                    identity_key: identity_key
+                        .parse::<IdentityKey>()
+                        .map(|ik| ik.to_proto())
+                        .map(Some)?,
+                });
+
+                // Instantiate an RPC client and send the request.
+                let GetValidatorInfoResponse { validator_info } = app
+                    .pd_channel()
+                    .await
+                    .map(StakeQueryServiceClient::new)?
+                    .get_validator_info(request)
+                    .await?
+                    .into_inner();
+
+                // Parse the validator status, or return an error if it was not found within the
+                // client's response.
+                let Info { status, .. } = validator_info
+                    .ok_or_else(|| anyhow!("response did not include validator info"))?
+                    .try_into()
+                    .context("parsing validator info")?;
+
+                // Serialize the status to TOML, and print it to stdout.
+                let toml = toml::to_string_pretty(&status)?;
+                println!("{}", toml);
+            }
```

Comment:
i think it would be maximally useful if it re-used the `pcli query validator list`, maybe adding a column for the validator's exchange rate:

```rust
// `U128x128` has a display impl
let validator_exchange_rate_bps_sq = U128x128::from(rate_data.validator_exchange_rate);
let validator_exchange_rate = (validator_exchange_rate_bps_sq / *BPS_SQUARED_SCALING_FACTOR).expect("nonzero scaling factor");
```

Reply:
that's an excellent idea.

i played around with this a bit this evening and found that re-using the existing rendering is going to be slightly difficult, due to the slightly different columns. (_we do not know the `share` column representing the percentage of voting power, because we only have information about one validator._)

i think this is doable, but i may drop the `status` subcommand from this particular PR and handle it in a subsequent PR. the change to `query validator definition` fixes a performance bug and isn't dependent on the addition of the `status` command.

Reply:
rebased onto main at 098d9cccd, dropped the `status` subcommand for a fast follow-on PR. :+1:

Reply:
Frankly I would prefer that we do this in this pr, and finish this ticket. The cumulative voting share does not matter. The reason this command is important is that we have no way to observe validator identities that are part of the main staking indexed since we merged #3846. My suggestion was about rendering the data in a table rather than pretty-printing a toml. The command is useful if we just render:
- validator info (name / IK)
- state
- validator exchange rate
- bonding state
- voting power

The rest is bonus and completely auxiliary. What matters is that there's some way to observe validator state. What we should tackle in a separate issue is discoverability of validator identities.

Reply:
thank you for the extra color. i've pushed 15954894c and 1b6c612ac, to reintroduce the `status` command, and to add logic to pretty-print the information in a table.